subsys/net/lib/rest_client/src/rest_client.c

```diff
@@ -293,6 +304,18 @@ static int rest_client_do_api_call(struct http_request *http_req,
 	resp_ctx->total_response_len = 0;
 	resp_ctx->used_socket_id = req_ctx->connect_socket;
 
+	if (req_ctx->timeout_ms > 0) {
+		/* Take time used for socket connect into account */
+		sckt_connect_time = k_uptime_get() - sckt_connect_start_time;
+
+		/* Check if timeout has already elapsed */
+		if (sckt_connect_time >= req_ctx->timeout_ms) {
+			LOG_WRN("Timeout occurred during socket connect");
+			return -ETIMEDOUT;
+		}
+		req_ctx->timeout_ms -= sckt_connect_time;
+	}
+
 	err = http_client_req(req_ctx->connect_socket, http_req, req_ctx->timeout_ms, resp_ctx);
```

Comment:
I do not think that passing zero as the timeout to http_client is allowed, no -- http_client will interpret this as "timeout immediately" (I confirmed this by looking at the source, not just reading its docs)

`SYS_FOREVER_MS` would be the correct value to pass for indefinite timeout. In my review I suggest that we change the entire location library to match with this standard. AFAICT, only one timer would need to be changed to achieve this

Reply:
> is comparison to K_FOREVER and K_NO_WAIT (=0) 

I am frankly not sure how I missed that, good point! Sorry about that! I stand completely corrected!

FWIW, I'm still glad you are switching to `int32_t` and `SYS_FOREVER_MS` instead of `uint16_t` and `0`, since I still think that is more in line with Zephyr nomenclature.

Reply:
I'll close this comment as we are moving to `int32_t` and `SYS_FOREVER_MS`.